src/python/nimbusml/tests/preprocessing/missing_values/test_data_with_missing.py

```diff
@@ -17,148 +18,19 @@
-        p = Pipeline([h, ft])
-        p.fit(df[['f', 'f1']].values, df['Label'])
-        res = p.predict(df[['f', 'f1']].values)
-        print(res)
-        print(p.summary())
-        assert_allclose(
-            res['Score'].values, [
-                4.965541, 0.519701, 4.992831, 3.877400, 5.020121], rtol=1e-4)
-
-    def test_input_conversion_to_float(self):
-        data={'f0': [0, 1, 2, 3],
-              'f1': [1, 2, 3, 4],
-              'f2': [1, 2, 3, 4],
-              'f3': [1, 2, 3, 4],
-              'f4': ['2', '3', '4', '5'],
-              'f5': [4, 5, np.nan, 9]}
-
-        data = DataFrame(data).astype({
-            'f0': np.int8,
-            'f1': np.int16,
-            'f2': np.int32,
-            'f3': np.int64,
-            'f4': str,
-            'f5': np.float64})
-
-        # Check Indicator
-        xf = Indicator()
-        result = xf.fit_transform(data)
-
-        assert_equal(result.loc[2, 'f5'], True)
-        result.loc[2, 'f5'] = False
-        result = ~result
-        self.assertTrue(result.all(axis=None))
-
-        # Check Filter
-        xf = Filter()
-        result = xf.fit_transform(data)
-        assert_equal(len(result), 3)
-        assert_equal(result.loc[2, 'f5'], 9.0)
-
-        # Check Handler
-        xf = Handler(replace_with='Mean')
-        result = xf.fit_transform(data)
-        assert_equal(len(result), 4)
-        assert_equal(result.loc[2, 'f5.f5'], 6.0)
-        assert_equal(result.loc[2, 'f5.IsMissing.f5'], 1.0)
-
-    def test_input_conversion_to_float_retains_other_column_types(self):
-        data={'f0': [0, 1, 2, 3],
-              'f1': ['2', '3', '4', '5'],
-              'f2': [4, 5, np.nan, 9]}
-
-        data = DataFrame(data).astype({
-            'f0': np.int32,
-            'f1': str,
-            'f2': np.float64})
-
-        # Check Indicator
-        xf = Indicator(columns={'f2.ind': 'f2'})
-        result = xf.fit_transform(data)
-        assert_equal(result.dtypes['f0'], np.int32)
-        assert_equal(result.dtypes['f1'], np.object)
-        assert_equal(result.dtypes['f2'], np.float64)
-        assert_equal(result.dtypes['f2.ind'], np.bool)
-        assert_equal(result.loc[2, 'f2.ind'], True)
-        assert_equal(len(result), 4)
-
-        # Check Filter
-        xf = Filter(columns=['f2'])
-        result = xf.fit_transform(data)
-        assert_equal(len(result), 3)
-        assert_equal(result.loc[2, 'f2'], 9.0)
-        assert_equal(result.dtypes['f0'], np.int32)
-        assert_equal(result.dtypes['f1'], np.object)
-        assert_equal(result.dtypes['f2'], np.float32)
-
-        xf = Filter(columns=['f1'])
+        # Check ToKeyImputer
+        xf = ToKeyImputer(columns={'f0.out': 'f0', 'f1.out': 'f1'})
         result = xf.fit_transform(data)
-        assert_equal(len(result), 4)
-        assert_equal(result.loc[3, 'f2'], 9.0)
-        assert_equal(result.dtypes['f0'], np.int32)
-        assert_equal(result.dtypes['f1'], np.float32)
```

Comment:
Put back deleted lines

Reply:
test

---
In reply to: [325808862](https://github.com/microsoft/NimbusML/pull/246#discussion_r325808862) [](ancestors = 325808862)